### DISCUSSION TITLE
RFC 7523: Support refresh token for specified client

### DISCUSSION THREAD

**Discussion by eonie** [ory/hydra#3290] [Q&A] [answered]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

In some scenarios，i need to get access token without a direct user-approval step at the Hydra server. For example , i want to integrate my existing application for user login and virify access token through Hydra. In the same time, i want to support Github Oauth2 for user who want to use.
So, I'm going to use the RFC7523 to get the access token when user passed the Github authentication。
Now the RFC7523 grant type only return bearer access token like this
```
{
    "access_token": "TIauaolbD9nUdAzLnpsLt9ljrxYhr0IbckRJMcFwkDU.NDGVZNOYD9XylETIi5uC5ZRdHcZGQ8vKM6qCKsDyXCE",
    "expires_in": 3599,
    "scope": "openid offline",
    "token_type": "bearer"
}
```
After the token expired, user will need to get a new access token.

### Describe your ideal solution

Based on the above description，i hope Hydra could return refresh token for client which has grant type `refresh_token`, and then the application could get a new `access token`  by the `refresh_token`.

### Workarounds or alternatives

None

### Version

1.11.10

**Comment by vinckr** [comment, marked as answer]
Hello @eonie 
Sorry for the late response, have you found a solution in the meantime? 

For integrating "Login with github" I would recommend using Ory Kratos, that implements all self-service user flows like signup, login etc. and also integrates nicely with OIDC compliant servers, e.g. Github https://www.ory.sh/docs/kratos/social-signin/github as well as Ory Hydra and solves your issue.

I will mark the discussion as answered now, but feel free to come back to this discussion at any time!
